### PR TITLE
Cache suite text

### PR DESCRIPTION
**Pull request by novakov-alexey**
Add Play Cache + integration tests

### CODE REVIEW

**Comment by novakov-alexey** [comment]
Could you review it? It is only 4 files changed.

File: server/app/controllers/TaskSolver.scala [hunk 1]
```diff
@@ -71,7 +71,7 @@ class TaskSolver @Inject()(executor: RuntimeSuiteExecutor with DynamicSuiteExecu
     }
   }
 
-  private def getTask(year: Long, taskType: String, timeuuid: String): Future[Option[Task]] = {
+  private def getCachedTask(year: Long, taskType: String, timeuuid: String): Future[Option[Task]] = {
```
Comment (novakov-alexey) [inline server/app/controllers/TaskSolver.scala]: getCachedTask name does not tell that function also work with database. I intentionally left it as getTask. But let's leave this new name for now.

Comment (DmytroOrlov) [inline server/app/controllers/TaskSolver.scala]: Let's rename it now or later. The reason to rename is next commit.

File: server/app/controllers/TaskSolver.scala [hunk 2]
```diff
@@ -72,24 +69,23 @@ class TaskSolver @Inject()(executor: RuntimeSuiteExecutor with DynamicSuiteExecu
...
+    val suiteKey = (year, taskType, timeuuid).toString()
 
-    Future (getFromCache) recoverWith {
-      case _ => getFromDb andThen saveInCache
+    val task = cache.get[Task](suiteKey)
+    task match {
+      case Some(_) => Future.successful(task)
+      case None =>
+        val f = getFromDb
+        f.foreach {
```
Comment (novakov-alexey) [inline server/app/controllers/TaskSolver.scala]: Is it possible to use map function instead of foreach?

Comment (DmytroOrlov) [inline server/app/controllers/TaskSolver.scala]: If we don't need the result of `map`, than let's avoid additional latency

**Pull request merged**